consensus/ethash/sealer.go

```diff
@@ -161,40 +164,65 @@ search:
 	runtime.KeepAlive(dataset)
 }
 
-// remote starts a standalone goroutine to handle remote mining related stuff.
-func (ethash *Ethash) remote() {
+// remote is a standalone goroutine to handle remote mining related stuff.
+func (ethash *Ethash) remote(notify []string) {
 	var (
-		works       = make(map[common.Hash]*types.Block)
-		rates       = make(map[common.Hash]hashrate)
-		currentWork *types.Block
+		works = make(map[common.Hash]*types.Block)
+		rates = make(map[common.Hash]hashrate)
+
+		currentBlock *types.Block
+		currentWork  [3]string
+
+		notifyTransport = &http.Transport{}
+		notifyClient    = &http.Client{
+			Transport: notifyTransport,
+			Timeout:   time.Second,
+		}
+		notifyReqs = make([]*http.Request, len(notify))
 	)
+	// notifyWork notifies all the specified mining endpoints of the availability of
+	// new work to be processed.
+	notifyWork := func() {
+		work := currentWork
+		blob, _ := json.Marshal(work)
 
-	// getWork returns a work package for external miner.
+		for i, url := range notify {
+			// Terminate any previously pending request and create the new work
+			if notifyReqs[i] != nil {
```

Comment:
Should we set the `notifyReq[i]` as nil if the request has been sent successfully?

Reply:
That would be racey. To be fair however, this current code might be racey too. I'll think about it.